components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -239,6 +269,42 @@ private <T> T deserialize(HttpEntity responseEntity, Class<T> returnType) throws
         }
     }
 
+    private String validateJsonResponse(HttpEntity responseEntity) throws ActionInvocationException {
+
+        if (!isAcceptablePayload(responseEntity)) {
+            throw new ActionInvocationException("The response content type is not application/json.");
+        }
+
+        try {
+            return EntityUtils.toString(responseEntity);
+        } catch (IOException e) {
+            throw new ActionInvocationException("Parsing JSON response failed.", e);
```

Comment:
This should rather say 'Reading http response failed.'

Reply:
This is fixed with 148073eab6305adc3e26c71ca755a0550e805f54